testing/python/approx.py

```diff
@@ -43,7 +44,256 @@ def report_failure(self, out, test, example, got):
     return MyDocTestRunner()
 
 
+@contextmanager
+def temporary_verbosity(config, verbosity=0):
+    original_verbosity = config.getoption("verbose")
+    config.option.verbose = verbosity
+    try:
+        yield
+    finally:
+        config.option.verbose = original_verbosity
+
+
+@pytest.fixture
+def assert_approx_raises_regex(pytestconfig):
+    def do_assert(lhs, rhs, expected_message, verbosity_level=0):
+        import re
+
+        with temporary_verbosity(pytestconfig, verbosity_level):
+            with pytest.raises(AssertionError) as e:
+                assert lhs == approx(rhs)
+
+        nl = "\n"
+        obtained_message = str(e.value).splitlines()
+        assert len(obtained_message) == len(expected_message), (
+            "Regex message length doesn't match obtained.\n"
+            "Obtained:\n"
+            f"{nl.join(obtained_message)}\n\n"
+            "Expected regex:\n"
+            f"{nl.join(expected_message)}\n\n"
+        )
+
+        for i, (obtained_line, expected_line) in enumerate(
+            zip(obtained_message, expected_message)
+        ):
+            regex = re.compile(expected_line)
+            assert regex.match(obtained_line) is not None, (
+                "Unexpected error message:\n"
+                f"{nl.join(obtained_message)}\n\n"
+                "Did not match regex:\n"
+                f"{nl.join(expected_message)}\n\n"
+                f"With verbosity level = {verbosity_level}, on line {i}"
+            )
+
+    return do_assert
+
+
+SOME_FLOAT = r"[+-]?([0-9]*[.])?[0-9]+\s*"
+
+
 class TestApprox:
+    def test_error_messages(self, assert_approx_raises_regex):
+        np = pytest.importorskip("numpy")
+
+        assert_approx_raises_regex(
+            2.0,
+            1.0,
+            [
+                "assert comparison failed",
```

Comment:
I believe the the purpose of the "assert" here is to show the fully resolved `<left> <op> <right>`, e.g.

```
x = 1
y = 2
assert x == y
```

shows

```
x.py:4: in test_x
    assert x == y
E   assert 1 == 2
```

So I think "assert comparison failed" is not what it's *supposed* to show.

Reply:
Hmm, I don't follow you on this one, sorry. You mean to change the test or the assertion message? :thinking:

Reply:
(got it now)